test/vtgatev2_test.py

```diff
@@ -1210,20 +1210,10 @@ def test_restart_mysql_tablet_queries(self):
       t.wait_for_vttablet_state('SERVING')
     self.replica_tablet2.kill_vttablet()
     replica_tablet_proc = self.replica_tablet.kill_vttablet(wait=False)
-    time.sleep(0.1)
+    if vtgate_gateway_flavor().flavor() == 'shardgateway':
+      time.sleep(1)  # skip the vttablet waiting period
     # send query while vttablet is in lameduck, should fail as no vttablet
-    try:
-      vtgate_conn._execute(
-          'select 1 from vt_insert_test', {},
-          KEYSPACE_NAME, 'replica',
-          keyranges=[self.keyrange])
-      self.fail('DatabaseError should have been raised')
-    except Exception, e:  # pylint: disable=broad-except
-      self.assertIsInstance(e, dbexceptions.DatabaseError)
-      self.assertNotIsInstance(e, dbexceptions.IntegrityError)
-      self.assertNotIsInstance(e, dbexceptions.OperationalError)
-      self.assertNotIsInstance(e, dbexceptions.TimeoutError)
-    # send another query, should also fail
+    time.sleep(0.1)  # wait a short while so vtgate gets the health check
```

Comment:
Here's an example of what I'm thinking of: https://github.com/youtube/vitess/blob/master/test/worker.py#L443

I don't expect that it'll help a lot for this particular place, but I think there's enough sleeps in this test that cleaning up many of them will help a lot.